thunder/torch/__init__.py

```diff
@@ -1503,6 +1503,39 @@ def exp2_(a):
     return prims.copy_(exp2(a), a)
 
 
+# fake out of place variant
+@torchsymbol(id="exponential")
+def exponential(a, rate=1, generator=None):
+    assert generator is None
+    utils.check(
+        not thunder.dtypes.is_complex_dtype(a.dtype)
+        and not thunder.dtypes.is_integer_dtype(a.dtype)
+        and not thunder.dtypes.is_boolean_dtype(a.dtype),
+        lambda: f"Exponential distribution is a continuous probability distribution. \
+        dtype must be a floating point but you specified {self.dtype}",
+    )
+    utils.check(
+        rate > 0.0,
+        lambda: f"exponential_ expects lambda > 0.0, but found lambda={rate}",
+    )
+    uniform_val = uniform_like(a)
+
+    # copying numerics of transformation::exponential see comment:
+    # curand_uniform has (0,1] bounds. log(1) is 0 and exponential excludes 0.
+    # we need log to be not 0, and not underflow when converted to half
+    # fast __logf approximation can underflow, so set log to -epsilon/2 for 1 or close to 1 args
+    epsilon = torch.finfo(thunder.dtypes.to_torch_dtype(a.dtype)).eps / 2
+    condition = uniform_val >= 1.0 - epsilon
+    log_uniform = where(condition, -epsilon, log(uniform_val))
+
+    return -1 / rate * log_uniform
+
+
+@torchsymbol(torch.Tensor.exponential_, id="exponential_", is_method=True, tags=(prims.OpTags.IN_PLACE,))
+def exponential_(a, rate=1, generator=None):
```

Comment:
Signature updates here, too, please (type annotations and positional-only and keyword-only arguments)